src/mumble/Settings.cpp

```diff
@@ -778,21 +780,7 @@ void Settings::load(QSettings* settings_ptr) {
 	SAVELOAD(bDirectInputVerboseLogging, "shortcut/windows/directinput/verboselogging");
 	SAVELOAD(bEnableUIAccess, "shortcut/windows/uiaccess/enable");
 
-	int nshorts = settings_ptr->beginReadArray(QLatin1String("shortcuts"));
-	for (int i=0; i<nshorts; i++) {
-		settings_ptr->setArrayIndex(i);
-		Shortcut s;
-
-		s.iIndex = -2;
-
-		SAVELOAD(s.iIndex, "index");
-		SAVELOAD(s.qlButtons, "keys");
-		SAVELOAD(s.bSuppress, "suppress");
-		s.qvData = settings_ptr->value(QLatin1String("data"));
-		if (s.iIndex >= -1)
-			qlShortcuts << s;
-	}
-	settings_ptr->endArray();
+	qlShortcuts = loadShortcuts(settings_ptr);
```

Comment:
Won't this fail to load the shortcuts now that the loadShortcuts function checks for fileVersion? The registry won't have that on first launch after update or am I missing something?

If that is the case my suggestion would be to have the loadShortcuts/saveShortcuts function be the normal load/save functionality and introduce a thin wrapper loadShortcutsFile/saveShortcutsFile that does the version safe/restore + file handling. That way we don't mix the file save/load specific logic with our normal registry operation while still preventing code-duplication.

Reply:
Yeah, I fucked that up…